### PR TITLE
fix clang-format Github action

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Backport of https://github.com/arangodb/arangodb/pull/16139

The clang-format check executed via Github actions suddenly started failing last week, without having made any changes to the Github action or the invoked Docker container. So Github seems to have changed something in their Github actions environment.
This PR repairs the Github action so that it doesn't fail with git errors anymore.
Relies on code in another repository: https://github.com/arangodb/clang-format-action/releases/tag/1.0.3

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [x] Backports
  - [x] Backport for 3.9: this PR
  - [ ] Backport for 3.8: -
  - [ ] Backport for 3.7: -

#### Related Information

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

**Review by neunhoef — APPROVED** [review]
LGTM

**Review by maierlars — APPROVED** [review]

File: arangod/Aql/SortExecutor.cpp [hunk 1]
```diff
@@ -38,8 +38,7 @@ using namespace arangodb;
 using namespace arangodb::aql;
 
 namespace {
-
-/// @brief OurLessThan
+// custom AqlValue-aware comparator for sorting
```
Comment (maierlars) [inline arangod/Aql/SortExecutor.cpp:41]: How is this related to the change?
